src/core/component.js

```diff
@@ -212,7 +215,7 @@ module.exports.registerComponent = function (name, definition) {
   });
 
   if (components[name]) {
-    throw new Error('The component ' + name + ' has been already registered');
+    warn('The component ' + name + ' has been already registered');
```

Comment:
A component registered twice it means that there are two versions of the same components or two different components with the same name colliding. That's a good reason to stop everything loudly and force the user to reevaluate.